src/modules/acbs/acbs-facility-guarantee.service.test.ts

```diff
@@ -116,10 +135,124 @@ describe('AcbsFacilityGuaranteeService', () => {
+
+      const createGuaranteeForFacilityPromise = service.createGuaranteeForFacility(facilityIdentifier, newFacilityGuarantee, idToken);
+
+      await expect(createGuaranteeForFacilityPromise).rejects.toBeInstanceOf(AcbsResourceNotFoundException);
+      await expect(createGuaranteeForFacilityPromise).rejects.toThrow(`Facility with identifier ${facilityIdentifier} was not found by ACBS.`);
+      await expect(createGuaranteeForFacilityPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws an AcbsBadRequestException if ACBS responds with a 400 that is a string that does not contain "The facility not found"', async () => {
+      const axiosError = new AxiosError();
+      const errorString = valueGenerator.string();
+      axiosError.response = {
+        data: errorString,
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      when(httpServicePost)
+        .calledWith(...expectedHttpServicePostArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const createGuaranteeForFacilityPromise = service.createGuaranteeForFacility(facilityIdentifier, newFacilityGuarantee, idToken);
+
+      await expect(createGuaranteeForFacilityPromise).rejects.toBeInstanceOf(AcbsBadRequestException);
+      await expect(createGuaranteeForFacilityPromise).rejects.toThrow(`Failed to create a guarantee for facility ${facilityIdentifier} in ACBS.`);
+      await expect(createGuaranteeForFacilityPromise).rejects.toHaveProperty('innerError', axiosError);
+      await expect(createGuaranteeForFacilityPromise).rejects.toHaveProperty('errorBody', errorString);
+    });
+
+    it('throws an AcbsBadRequestException if ACBS responds with a 400 that is not a string', async () => {
+      const axiosError = new AxiosError();
+      const errorBody = { errorMessage: valueGenerator.string() };
+      axiosError.response = {
+        data: errorBody,
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      when(httpServicePost)
+        .calledWith(...expectedHttpServicePostArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const createGuaranteeForFacilityPromise = service.createGuaranteeForFacility(facilityIdentifier, newFacilityGuarantee, idToken);
+
+      await expect(createGuaranteeForFacilityPromise).rejects.toBeInstanceOf(AcbsBadRequestException);
+      await expect(createGuaranteeForFacilityPromise).rejects.toThrow(`Failed to create a guarantee for facility ${facilityIdentifier} in ACBS.`);
+      await expect(createGuaranteeForFacilityPromise).rejects.toHaveProperty('innerError', axiosError);
+      await expect(createGuaranteeForFacilityPromise).rejects.toHaveProperty('errorBody', JSON.stringify(errorBody));
+    });
+
+    it('throws an AcbsUnexpectedException if ACBS responds with an error code that is not 400', async () => {
```

Comment:
Can we add some assertions with various payload fields with void, invalid and out of bounds values please.